crypto/cms/cms_sd.c

```diff
@@ -372,7 +372,7 @@ CMS_SignerInfo *CMS_add1_signer(CMS_ContentInfo *cms,
             ESS_SIGNING_CERT_V2 *sc2 = NULL;
             int add_sc;
 
-            if (md == EVP_sha1() || md == NULL) {
+            if (md == NULL || EVP_MD_is_a(md, SN_sha1)) {
```

Comment:
Shouldn't EVP_MD_is_a() check if the passed method is NULL, and always return false in that case?

Reply:
This is not a blocker, it can be dealt with in another PR